### DISCUSSION TITLE
Backrest Windows SFTP Repositorie

### DISCUSSION THREAD

**Discussion by ttoopp** [garethgeorge/backrest#593] [General]
Hi, 

I try to connect to a SFTP via the Backrest GUI. 
I have generated a Publickey with Putty and stored it on the server.
With Putty and the private key, I can log on to my server without a password.

Now I have tried to create a repo with Backrest. I have set the following URI and flags.

`URI: sftp:backupuser@backup01.hostname.com:/backup`

`-o sftp.args="-i C:\Users\aa\Desktop\backup.ppk"
`
When I click on create, the circle at the bottom spins all the time and nothing happens.

**Comment by maelstrm-code** [comment]
Sharing after fighting with this today as a newbie:

I think Putty uses a different format for private keys. You might need to use the ssl-keygen in windows terminal then refer to those keys.

eg:
https://www.howtogeek.com/762863/how-to-generate-ssh-keys-in-windows-10-and-windows-11/

Note - might want to use a different cypher eg check out generating using the flag
-t ed25519

**Comment by raphpa** [comment]
You cannot use the .ppk putty key format indeed. But you can load that key back into `puttyGen` Putty Key Generator and go to `Conversions` -> `Export OpenSSH key`. That one should work then.
